server/src/main/java/org/elasticsearch/index/mapper/RangeType.java

```diff
@@ -107,9 +107,8 @@ public BytesRef encodeRanges(Set<RangeFieldMapper.Range> ranges) throws IOExcept
         }
 
         @Override
-        public List<RangeFieldMapper.Range> decodeRanges(BytesRef bytes) {
-            // TODO: Implement this.
-            throw new UnsupportedOperationException();
+        public List<RangeFieldMapper.Range> decodeRanges(BytesRef bytes) throws IOException {
```

Comment:
This is an interesting piece. I don't understand why it was not implemented given that `BinaryRangeUtil#decodeIPRanges` was introduced as a part of the same change. It works with this impl though.

Reply:
I see method `decodeRange` is used in our `histogram` aggregation...I guess this means it is not possible to do an histogram aggregation on an `ip_range` field while it is (apparently) for dates and longs? I don't know why TBH. I was wondering if it would be better to add support for this in another PR so that we can test it against aggregations which are supposed to be supported and also to make sure we are BWC.

Reply:
Yes, ip ranges would hit this https://github.com/elastic/elasticsearch/blob/0075c1fb1e82c51c747729924ce67efef6daadeb/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/RangeHistogramAggregator.java#L67 and so are explicitly not supported. I'll open a separate PR.

Reply:
Opened #107376.